### ISSUE TITLE
torch.autograd.set_detect_anomaly(True) does not exist in C++?

### ISSUE THREAD

**Issue by JesperLindberg** [pytorch/pytorch#35872]
## 🐛 Bug

I got the following error when calling `optimizer.backward()`.
`one of the variables needed for gradient computation has been modified by an inplace operation: [CPUFloatType [50, 40]], which is output 0 of TBackward, is at version 2; expected version 1 instead. Hint: enable anomaly detection to find the operation that failed to compute its gradient, with torch.autograd.set_detect_anomaly(True). after train`

## To Reproduce
Not sure, since I can't see which operation that makes my code crash.

## Expected behavior

I expected the function that was listed in the error to exist.

## Environment

 - PyTorch Version (e.g., 1.0): Libtorch 1.4
 - OS (e.g., Linux): Windows 7 64 bit
 - How you installed PyTorch (`conda`, `pip`, source): source
 - Build command you used (if compiling from source): none
 - Python version: 3.6
 - CUDA/cuDNN version: none
 - GPU models and configuration: none
 - Any other relevant information: none

cc @yf225

**Comment by yf225** [comment]
`torch::autograd::set_detect_anomaly(bool)` doesn't exist in the C++ frontend yet. cc. @glaringlee  it might be something we can consider adding

**Comment by glaringlee** [comment]
Assign to myself for now. Will prioritize this work later.

**Comment by payymann** [comment]
Any update in this issue or we most forget the c++ frontend? o|O

**Comment by taras-janea** [comment]
The issue is resolved by #46981, which added anomaly detection mode to the C++ API. You can now enable it in LibTorch with:
```
torch::autograd::AnomalyMode::set_enabled(true);
```
This is the C++ equivalent of `torch.autograd.set_detect_anomaly(True)` in Python, and will provide stack traces to help debug in-place operation errors like the one described here.
